From 7035becb0f8485bf71e934a36ca3f3673a0a880e Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:32:14 -0400
Subject: [scsi] hpsa: do not require board "not ready" status after hard reset

Message-id: <1410186734-19986-84-git-send-email-thenzl@redhat.com>
Patchwork-id: 92093
O-Subject: [RHEL6.6 e-stor PATCH 83/83] hpsa: do not require board "not ready" status after hard reset
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Immediately following a hard board reset, There are some
mandatory delays during which we must not access the board
and during which we might miss the "not ready" status,
therefore it is a mistake to look for and expect to see
the "not ready" status.

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 0ddf1d77505fd97daa7fc70bb0498c54a5c33725)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 3efaf10..8d044e2 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -6041,23 +6041,6 @@ static int hpsa_kdump_hard_reset_controller(struct pci_dev *pdev)
     need a little pause here */
  msleep(HPSA_POST_RESET_PAUSE_MSECS);
 
- if (!use_doorbell) {
-  /* Wait for board to become not ready, then ready.
-   * (if we used the doorbell, then we already waited 5 secs
-   * so the "not ready" state is already gone by so we
-   * won't catch it.)
-   */
-  dev_info(&pdev->dev, "Waiting for board to reset.\n");
-  rc = hpsa_wait_for_board_state(pdev, vaddr, BOARD_NOT_READY);
-  if (rc) {
-   dev_warn(&pdev->dev,
-    "failed waiting for board to reset."
-    " Will try soft reset.\n");
-   /* Not expected, but try soft reset later */
-   rc = -ENOTSUPP;
-   goto unmap_cfgtable;
-  }
- }
  rc = hpsa_wait_for_board_state(pdev, vaddr, BOARD_READY);
  if (rc) {
   dev_warn(&pdev->dev,
-- 
1.7.1